.github/workflows/router-e2e.yml

```diff
@@ -0,0 +1,71 @@
+name: Router Maestro E2E
+
+on:
+  pull_request:
+    paths:
+      - packages/@expo/cli/**
+      - packages/expo-router/**
+
+jobs:
+  router-maestro:
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Use Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '20.x'
+
+      - name: Install dependencies
+        run: YARN_IGNORE_SCRIPTS=true yarn --frozen-lockfile
+
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'temurin'
+          java-version: '17'
+
+      - name: Install Maestro
+        run: |
+          export MAESTRO_VERSION=1.36.0
+          curl -Ls "https://get.maestro.mobile.dev" | bash
+          arch -arm64 brew tap facebook/fb
+          arch -arm64 brew install facebook/fb/idb-companion
+          echo "$HOME/.maestro/bin" >> $GITHUB_PATH
+
+      - name: Start iOS Simulator
+        run: |
+          # Find a simulator by name (e.g., iPhone 15)
+          DEVICE_NAME="iPhone 15 Pro"
```

Comment:
This seems fragile, maestro has the ability to attach to any simulator. We should probably just start any sim to ensure the test doesn't break during xcode updates.